tests/ocean/test_compute_flow.py

```diff
@@ -156,13 +157,19 @@ def run_compute_test(
 
     status = ocean_instance.compute.status(did, job_id, consumer_wallet)
     print(f"got job status: {status}")
-    assert (
-        status and status["ok"]
-    ), f"something not right about the compute job, got status: {status}"
-
-    status = ocean_instance.compute.stop(did, job_id, consumer_wallet)
-    print(f"got job status after requesting stop: {status}")
-    assert status, f"something not right about the compute job, got status: {status}"
+    if not expect_failure:
+        assert (
+            status and status["ok"]
+        ), f"something not right about the compute job, got status: {status}"
+
+        status = ocean_instance.compute.stop(did, job_id, consumer_wallet)
+        print(f"got job status after requesting stop: {status}")
+        assert (
+            status
+        ), f"something not right about the compute job, got status: {status}"
+    else:
+        assert status["status"] == 400
```

Comment:
I am a big fan of "if" inversion. I'd treat the expect_failure case first, then add a `return` line. I'd then go on with the flow that was previously in place, without the need for an else branch (return makes it moot).

I would also change the assert message to something more suited, e.g. "request expected 400 error but got status:" ...